3761/1/top_gems.rb

```diff
@@ -1,29 +1,27 @@
-require './parser_argv.rb'
-require './initial_list_of_gems.rb'
+require './argv_parser.rb'
+require './gem_list_creator.rb'
 require './page.rb'
 require 'mechanize'
-require './table.rb'
-require './gem_statisic_fetcher.rb'
-parser = ParserArgv.new(ARGV)
-file_name = parser.file_name
-file_name ||= '.top-gems.yml'
-gems = InitialListOfGems.new(file_name)
-list_gems = if parser.part_name_gem.nil?
-              gems.list_gems
-            else
-              gems.sort(parser.part_name_gem)
-            end
+require 'terminal-table'
+require './gem_statistic_fetcher.rb'
+require './gem_statistic_analyzer.rb'
+require './gem_serializer.rb'
+parser = ArgvParser.new(ARGV)
+parser.call
+gem_list = GemListCreator.new(parser.file_name, parser.part_name_gem).call
 agent = Mechanize.new
-
-array_gems_score = list_gems.map do |gem_name|
-  GemStatisicFetcher.fetch_statistic(gem_name, agent)
+array_gems_score = gem_list.map do |gem_name|
+  stats = GemStatisticFetcher.new(gem_name, agent).call
+  GemStatisticAnalyzer.new(stats).call
 end
 
-# require 'pry'
-# binding.pry
-array_gems_score.sort_by! { |hash| hash[:score] }.reverse!
-count = list_gems.count
-top = parser.top_count
-top ||= count
-table = Table.new(array_gems_score, top)
-table.write
+array_gems_score.sort! { |hash_a, hash_b| hash_b[:score] <=> hash_a[:score] }
+
+table = Terminal::Table.new
+row_count = parser.top_count || array_gems_score.count
+row_count = array_gems_score.count if row_count > array_gems_score.count
```

Comment:
merge this 2 lines

Reply:
How?)

Reply:
Try something like this:
```ruby
row_count = [parser.top_count, array_gems_score.count].min
```
Better to find your own way so try it by urself